cdap-gateway/src/main/java/co/cask/cdap/gateway/router/handlers/OutboundHandler.java

```diff
@@ -48,7 +48,16 @@ public void messageReceived(final ChannelHandlerContext ctx, final MessageEvent
     Channels.write(inboundChannel, msg).addListener(new ChannelFutureListener() {
       @Override
       public void operationComplete(ChannelFuture future) throws Exception {
-        OutboundHandler.super.messageReceived(ctx, e);
+        ctx.getPipeline().execute(new Runnable() {
+          @Override
+          public void run() {
+            try {
+              OutboundHandler.super.messageReceived(ctx, e);
+            } catch (Exception ex) {
+              LOG.error("Got exception {}", ctx.getChannel(), ex);
```

Comment:
Can this be a better message describing possible problem rather than letting user decipher this cryptic stack here.